src/parser.rs

```diff
@@ -142,6 +156,45 @@ impl From<&Graph> for WeightedGraph {
     }
 }
 
+impl AdjacencyMatrix for Graph {
```

Comment:
Do we `really` want to offer this?

I feel like most most algorithms are optimized for either representation, matrix or list. But pretending that a Graph is a Adjacency**Matrix** seems like an invitation to write bad code.

Why do you want this? Especially my exact algorithms where I am doing that many cycle traversals are really really not made for adjacancy lists.

Reply:
Ok, that is a valid point. I removed it.